proxy/deployments/guardrails/README.md

```diff
@@ -0,0 +1,214 @@
+# Guardrails
+
+Guardrails is a security feature designed specifically for OPEA RAG pipelines. Guardrails helps eliminate hallucination and toxicity, and build trust from end users.
+
+Guardrails in OPEA Pipeline Proxy supports both rule-based detection and AI-powered content safety. You can leverage rule-based detection to design keyword filter to avoid the leakage of PII (personal identifiable information) or choose a SPAM model to filter SPAMs out.
+
+## Architecture
+
+```mermaid
+flowchart LR
+    subgraph Filter[HTTP Filter]
+        direction BT
+        Guardrails <--> OpenVINO
+    end
+    Model --> Filter
+    Re(Request / Response) --> Filter
+    Filter --> Continue(Continue)
+    Filter --> Stop(Stop)
+```
+
+## Deployment
+
+The tutorial introduces how to leverage Guardrails and filter out SPAMs in requests on the ingress gateway for a ChatQnA pipeline.
+
+### Preparing the model used in Guardrails
+
+OPEA Pipeline Proxy leverages models in OpenVINO IR format. You can convert different models into OpenVINO IR format following [these instructions](https://docs.openvino.ai/2024/openvino-workflow/model-preparation/convert-model-to-ir.html).
+
+A Transformer model is usually consists of a tokenizer and some Transformer blocks. In OpenVINO IR format, the tokenizer and blocks can be merged into a file. You can merge the tokenizer and the model following [these instructions](https://docs.openvino.ai/2024/notebooks/openvino-tokenizers-with-output.html#merge-tokenizer-into-a-model).
+
+In the tutorial, we use [Titeiiko/OTIS-Official-Spam-Model](https://huggingface.co/Titeiiko/OTIS-Official-Spam-Model) for SPAM filtering.
+
+```sh
+# Convert HuggingFace model to OpenVINO IR.
+optimum-cli export openvino --model Titeiiko/OTIS-Official-Spam-Model OTIS-Official-Spam-Model
+# Convert the tokenizer to OpenVINO IR.
+convert_tokenizer Titeiiko/OTIS-Official-Spam-Model -o ./OTIS-Official-Spam-Model
+```
+
+```python
+from openvino import Core, save_model
+from openvino_tokenizers import connect_models
+
+core = Core()
+tokenizer = core.read_model("OTIS-Official-Spam-Model/openvino_tokenizer.xml")
+model = core.read_model("OTIS-Official-Spam-Model/openvino_model.xml")
+# Merge the model and its tokenizer.
+combined_model = connect_models(tokenizer, model)
+save_model(combined_model, "OTIS-Official-Spam-Model/combined.xml")
+```
+
+### Mounting models in the container
+
+To leverage models, you have to mount them to the OPEA Pipeline Proxy container. Please follow the field [here](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec) to mount the model into the ingress gateway pod.
```

Comment:
field? 
s/Please follow the field [here] (url)/Follow the instructions at <url>